awx/main/models/ha.py

```diff
@@ -161,6 +161,24 @@ def consumed_capacity(self):
     def remaining_capacity(self):
         return self.capacity - self.consumed_capacity
 
+    @staticmethod
+    def get_remaining_capacity(instances={}, jobs=[]):
+        """Takes mapping of hostname to SimpleNamespace instance like objects and a list of jobs.
+
+        Computes remaining capacity for all the instances.
+
+        Updates the "remaining_capacity" field on the SimpleNamespace instance like object.
+        For use in the task manager to avoid refetching jobs from the database.
+        """
+        for job in jobs:
```

Comment:
You should `continue` if the job's status is not something that counts toward capacity usage. You should double-check, but I believe that only "running" and "waiting" jobs count.

Reply:
yeah I'll do that when I start passing all the jobs (I had just passed already selected running/waiting jobs)